Platon/backend/tests/follow_test.py

```diff
@@ -217,3 +217,85 @@ def test_unfollow(self):
     def tearDown(self):
         super().tearDown()
+
+class ReportTest(BaseTest):
+    """
+        Unit Tests of the report module
+    """
+    def setUp(self):
+
+        jobs = [
+            Jobs("academician"),
+            Jobs("PhD student")
+        ]
+
+        for job in jobs:
+            db.session.add(job)
+
+        db.session.commit()
+
+        # Umut and Can are public users. Alperen is private user.
+        users = [
+            User("umut@deneme.com", True, "b73ec5e4625ffcb6d0d70826f33be7a75d45b37046e26c4b60d9111266d70e32", 3.5,
+                 "Umut", "Özdemir", False, None, None, None, 1, "boun"),
+            User("can@deneme.com", True, "cce0c2170d1ae52e099c716165d80119ee36840e3252e57f2b2b4d6bb111d8a5", 3.4,
+                 "Can", "Deneme", False, None, None, None, 2, "boun"),
+            User("alperen@deneme.com", True, "hashedpassword", 4.6, "Alperen", "Ozprivate", True, None, None, None, 1, "boun"),
+            User("hilal@deneme.com", True, "hasheddpassword", 4.5, "Hilal", "Private", True, None, None, None, 1, "boun")
+        ]
+        for user in users:
+            db.session.add(user)
+
+        db.session.commit()
+
+        db.session.commit()
+
+        collaborations = [
+            Collaboration(user_1_id=1, user_2_id=2),
+            Collaboration(user_1_id=2, user_2_id=1)
+        ]
+
+        for collaboration in collaborations:
+            db.session.add(collaboration)
+        db.session.commit()
+
+        # Add artificial users to test follow feature.
+        follows = [
+            Follow(1, 2),  # Umut follows Can
+            Follow(3, 2),  # Alperen follows Can
+            Follow(3, 1)  # Alperen follows Umut
+        ]
+        for follow in follows:
+            db.session.add(follow)
+
+
+    def test_post_report_valid(self):
+        valid_token = generate_token(1, datetime.timedelta(minutes=10))
+        data = {
+            'reported_user_id': 2,
+            'text': "report"
+        }
+        actual_response = self.client.post('/api/follow/report',data=data,headers={'auth_token': valid_token})
+
+        self.assertEqual(actual_response.status_code, 200, 'Incorrect HTTP Response Code')
+
+    def test_post_report_invalid(self):
+        valid_token = generate_token(1, datetime.timedelta(minutes=10))
+        data = {
+            'reported_user_id': 3,
+            'text': "report"
+        }
+        actual_response = self.client.post('/api/follow/report',data=data,headers={'auth_token': valid_token})
+
+        self.assertEqual(actual_response.status_code, 403, 'Incorrect HTTP Response Code')
+
+    def test_get_reports(self):
+        valid_token = generate_token(1, datetime.timedelta(minutes=10))
+
+        actual_response = self.client.get('/api/follow/report', query_string={'reported_user_id': 2},
+                                          headers={'auth_token': valid_token})
+
+        self.assertEqual(actual_response.status_code, 200, 'Incorrect HTTP Response Code')
```

Comment:
Adding a test for asserting response JSON data is equal to expected JSON data would be better.